### ISSUE TITLE
FIPS Country and Region Codes Are Not Standard!

### ISSUE THREAD

**Issue by o6uoq** [dynatrace-oss/terraform-provider-dynatrace#601]
Dynatrace states that they use [FIPS 10-4 codes,](https://en.wikipedia.org/wiki/List_of_FIPS_region_codes) but the API responses and API working logic says otherwise.

Where it says FIPS 10-4 is used:

- https://docs.dynatrace.com/docs/discover-dynatrace/references/dynatrace-api/environment-api/settings/schemas/builtin-rum-ip-mappings
- https://registry.terraform.io/providers/dynatrace-oss/dynatrace/1.32.0/docs/resources/synthetic_location
- https://www.pulumi.com/registry/packages/dynatrace/api-docs/rumiplocations/

That _should_ imply that countries like Australia are *AS*, but *AU* is actually the correct country code respected by Dynatrace.

There's some use of FIPS e.g. region codes:
```
❯ curl -s http://efele.net/maps/fips-10/data/fips-414.txt | grep ^AS   
AS00_414_414_country____AUSTRALIA__
AS01_414_414_territory____Australian Capital Territory__
AS02_414_414_state____New South Wales__
AS03_414_414_territory____Northern Territory__
AS04_414_414_state____Queensland__
AS05_414_414_state____South Australia__
AS06_414_414_state____Tasmania__
AS07_414_414_state____Victoria__
AS08_414_414_state____Western Australia__
```

..but the same does not apply for all countries e.g:

```
❯ curl -s http://efele.net/maps/fips-10/data/fips-414.txt | grep ^AS   
AS00_414_414_country____AUSTRALIA__
AS01_414_414_territory____Australian Capital Territory__
AS02_414_414_state____New South Wales__
AS03_414_414_territory____Northern Territory__
AS04_414_414_state____Queensland__
AS05_414_414_state____South Australia__
AS06_414_414_state____Tasmania__
AS07_414_414_state____Victoria__
AS08_414_414_state____Western Australia__
```

Another example is London aka City of London, where they have come up with LND I have NFI.

The drift between Dynatrace documentation, Dynatrace APIs and reality is embarrassing. It circumvents standards, standards in which you proclaim to use, yet no documentation to validate the cross-breeding, hybrid mixmash that is how they currently define country + region codes.

**Comment by Reinhard-Pilz-Dynatrace** [comment]
Hi @o6uoq 

It appears that we (from the team developing the Terraform Provider) have missed out on an important change regarding Region Codes and Country Codes for that specific REST API.

That change was as a matter of fact hard to track down when following the official documentation.
But it became very apparent once I took a look at the Open API Schema of the relevant endpoint.

The REST API meanwhile offers dedicated endpoints for the [supported country codes](https://dt-url.net/37030go) and [region codes](https://dt-url.net/az230x0). While both of them are still based on the two standards mentioned in our docs, they are different in many cases.

On short notice we will be updating the documentation of these attributes in our Terraform Provider (coming with the next release).

As an additional task I've scheduled in #604. With these Data Sources you won't have to manually look up valid codes via REST API.
